src/numeric-glib-1.0.vapi

```diff
@@ -92,18 +92,52 @@ namespace Numeric
 				_quadmath_snprintf (buffer, format, this);
 			}
 			return (string) buffer;
-
 		}
 	}
 
 	[FloatingType (rank = 6)]
-	public struct decimal32 {}
+	public struct decimal32
+	{
+		[CCode (cname = "strtod32", cheader_filename="dfp/stdlib.h")]
+		public static decimal32 parse (string s, out char sp = null);
+		public string to_string ()
+		{
+			return "%Hf".printf (this);
```

Comment:
This code is not going to work because GLib has its own print formatting implementation. We should check if we can use the printf-hook from libdfp in GLib.